zerver/tests/test_users.py

```diff
@@ -414,6 +414,21 @@ def test_allowed_format_complex(self) -> None:
         result = self.client_patch("/json/users/{}".format(self.example_user("hamlet").id), req)
         self.assert_json_success(result)
 
+    def test_require_unique_names(self) -> None:
+        new_name = "IaGo"
+        self.login("iago")
+        hamlet = self.example_user("hamlet")
+
+        do_set_realm_property(hamlet.realm, "require_unique_names", True, acting_user=None)
+        req = dict(full_name=new_name)
+        result = self.client_patch(f"/json/users/{hamlet.id}", req)
+        self.assert_json_error(result, "User name already exists!")
+
+        do_set_realm_property(hamlet.realm, "require_unique_names", False, acting_user=None)
+        req = dict(full_name=new_name)
+        result = self.client_patch(f"/json/users/{hamlet.id}", req)
+        self.assert_json_success(result)
```

Comment:
You should add a test case where the unique normalization part is relevant.

Reply:
done!